dbms/src/Server/Server.cpp

```diff
@@ -1085,6 +1085,13 @@ int Server::main(const std::vector<std::string> & /*args*/)
       *  settings, available functions, data types, aggregate functions, databases...
       */
     global_context = Context::createGlobal();
+    SCOPE_EXIT({
+        if (!proxy_conf.is_proxy_runnable)
+            return;
+
+        LOG_INFO(log, "Unlink tiflash_instance_wrap.tmt");
+        tiflash_instance_wrap.tmt = nullptr;
```

Comment:
```suggestion
        // Reset the `tiflash_instance_wrap.tmt` before `global_context` get released, or it will be a dangling pointer
        tiflash_instance_wrap.tmt = nullptr;
```